devito/types/dimension.py

```diff
@@ -716,89 +801,17 @@ def bound_symbols(self):
-            val = symbolic_thickness.subs(self._thickness_map)
-            return SubDimensionOffset(
-                int(val),
-                self.parent.symbolic_max,
-                symbolic_thickness
-            )
-
-    @cached_property
-    def _offset_right(self):
-        # The right extreme of the SubDimension can be related to either the
-        # min or max of the parent dimension
-        try:
-            symbolic_thickness = self.symbolic_max - self.parent.symbolic_min
-            val = symbolic_thickness.subs(self._thickness_map)
-            return SubDimensionOffset(
-                int(val),
-                self.parent.symbolic_min,
-                symbolic_thickness
-            )
-        except TypeError:
-            symbolic_thickness = self.symbolic_max - self.parent.symbolic_max
-            val = symbolic_thickness.subs(self._thickness_map)
-            return SubDimensionOffset(
-                int(val),
-                self.parent.symbolic_max,
-                symbolic_thickness
-            )
-
     @property
     def _arg_names(self):
-        return tuple(k.name for k, _ in self.thickness) + self.parent._arg_names
+        return tuple(k.name for k in self.thickness) + self.parent._arg_names
 
     def _arg_defaults(self, grid=None, **kwargs):
         return {}
 
     def _arg_values(self, interval, grid=None, **kwargs):
-        # Allow override of thickness values to disable BCs
-        # However, arguments from the user are considered global
-        # So overriding the thickness to a nonzero value should not cause
-        # boundaries to exist between ranks where they did not before
-        r_ltkn, r_rtkn = (
-            kwargs.get(k.name, v) for k, v in self.thickness
-        )
-
-        if grid is not None and grid.is_distributed(self.root):
-            # Get local thickness
-            if self.local:
-                # dimension is of type ``left``/right`` - compute the 'offset'
-                # and then add 1 to get the appropriate thickness
-                if r_ltkn is not None:
-                    ltkn = grid.distributor.glb_to_loc(self.root, r_ltkn-1, LEFT)
-                    ltkn = ltkn+1 if ltkn is not None else 0
-                else:
-                    ltkn = 0
-
-                if r_rtkn is not None:
-                    rtkn = grid.distributor.glb_to_loc(self.root, r_rtkn-1, RIGHT)
-                    rtkn = rtkn+1 if rtkn is not None else 0
-                else:
-                    rtkn = 0
-            else:
-                # dimension is of type ``middle``
-                ltkn = grid.distributor.glb_to_loc(self.root, r_ltkn, LEFT) or 0
-                rtkn = grid.distributor.glb_to_loc(self.root, r_rtkn, RIGHT) or 0
-        else:
-            ltkn = r_ltkn or 0
-            rtkn = r_rtkn or 0
-
-        return {i.name: v for i, v in zip(self._thickness_map, (ltkn, rtkn))}
+        # SubDimension thicknesses at runtime are calculated by the thicknesses
```

Comment:
as I wrote in another comment, this SubDimension should trigger the processing of its own arguments, namely its own Thicknesses

Reply:
They shouldn't (or at least it would be redundant if it did), see my other comment

Reply:
gotcha